src/Generate.purs

```diff
@@ -225,11 +225,22 @@ printResults :: Array FetchResult -> String
 printResults xs = replace { from: "INPUTS", to: inputs } template
   where inputs = Array.foldMap printResult xs
 
-generate :: Aff Unit
-generate = do
+concatMapM :: forall a b m. Monad m => (a -> m (Array b)) -> Array a -> m (Array b)
+concatMapM f xs = Array.concat <$> traverse f xs
+
+chunk :: forall a. Int -> Array a -> Array (Array a)
+chunk _ [] = []
```

Comment:
you should check what code this generates, as it might be worse than using array.null

Reply:
I'm not exactly sure what you mean, but if you mean changing the code to read like this:

```
chunk n xs | Array.null xs = []
           | n < 1         = [xs]
           | otherwise     = chunk' [] xs
```

then the generated code diff, modulo renaming, boils down to this:

```
@@ -7475,38 +7479,38 @@ var PS = {};
           };
       };
   };
-  var chunk = function (v) {
-      return function (v1) {
-          if (v1.length === 0) {
+  var chunk = function (n) {
+      return function (xs) {
+          if (Data_Array["null"](xs)) {
               return [  ];
           };

```

and if I'm reading that right, it's probably less efficient since `Array.null` is implemented as:

https://github.com/purescript/purescript-arrays/blob/463dcacb99455de5e28ac4a3312bb795f293d2d9/src/Data/Array.purs#L213

But maybe I misunderstand what you mean.